### ISSUE TITLE
Dependency conflict: flask, rasterio, click

### ISSUE THREAD

**Issue by soxofaan** [Open-EO/openeo-geopyspark-driver#694]
latest builds fail because of dependency conflict

```
ERROR: Cannot install openeo-driver, openeo-geopyspark and rasterio because these package versions have conflicting dependencies.

The conflict is caused by:
    rasterio 1.1.8 depends on click<8 and >=4.0
    cligj 0.5.0 depends on click<8 and >=4.0
    flask 3.0.2 depends on click>=8.1.3
    rasterio 1.1.8 depends on click<8 and >=4.0
    cligj 0.5.0 depends on click<8 and >=4.0
    flask 3.0.1 depends on click>=8.1.3
    rasterio 1.1.8 depends on click<8 and >=4.0
    cligj 0.5.0 depends on click<8 and >=4.0
    flask 3.0.0 depends on click>=8.1.3
    rasterio 1.1.8 depends on click<8 and >=4.0
    cligj 0.5.0 depends on click<8 and >=4.0
    flask 2.3.3 depends on click>=8.1.3
    rasterio 1.1.8 depends on click<8 and >=4.0
    cligj 0.5.0 depends on click<8 and >=4.0
    flask 2.3.2 depends on click>=8.1.3
    rasterio 1.1.8 depends on click<8 and >=4.0
    cligj 0.5.0 depends on click<8 and >=4.0
    flask 2.3.1 depends on click>=8.1.3
    rasterio 1.1.8 depends on click<8 and >=4.0
    cligj 0.5.0 depends on click<8 and >=4.0
    flask 2.3.0 depends on click>=8.1.3
    rasterio 1.1.8 depends on click<8 and >=4.0
    cligj 0.5.0 depends on click<8 and >=4.0
    flask 2.2.5 depends on click>=8.0
```

**Comment by soxofaan** [comment]
In https://github.com/Open-EO/openeo-python-driver/commit/3c98185fb0c357fe88911e3f6ddb9097d07b0afd  I introduced  a lower bound for flask: `>=2.2.5`, which I thought was reasonable (released 2023-05-02)

but apparently that now gives a conflict with 
https://github.com/Open-EO/openeo-geopyspark-driver/blob/9a8bb41a4be18693ba76334387212591a2579c75/setup.py#L63

flask 2.2.5 requires click>=8, while rasterio 1.1.8 does not allow click 8

**Comment by soxofaan** [comment]
I'll quick-fix it by lowering the flask requirement in openeo-python-driver

**Comment by soxofaan** [comment]
latest build that works apparently resolves to flask-2.0.3, which is pretty old (2022-02-14)

**Comment by soxofaan** [comment]
a better fix is getting rid of the `rasterio~=1.1.8` pin (a release from oct 2020)

**Comment by soxofaan** [comment]
env build now works again

closing for now
